src/scripts/main.js

```diff
@@ -18,10 +18,40 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null) {
+    throw new Error(`First argument must not be null or undefined`);
+  }
+
+  if (data === undefined) {
+    throw new Error(`Second argument must not be null or undefined`);
+  }
+
+  if (Object.entries(data).length === 0) {
+    return;
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const [key, value] of Object.entries(data)) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(value).length > 0) {
+      createTree(li, value);
+    }
+    ul.appendChild(li);
+  }
+
+  element.appendChild(ul);
 }
 
 createTree(tree, food);
```

Comment:
This line of code is fine, but it's better to separate the logic of creating a tree from the logic of appending it to the DOM. You could create a separate function for appending the tree to the DOM, and call it after the tree is created.